src/main/java/bio/terra/app/logging/LoggerInterceptor.java

```diff
@@ -39,6 +46,20 @@ public void afterCompletion(
 
     // skip logging the status endpoint
     if (!url.endsWith("/status")) {
+
+      Map<String, Object> stackDriverPayload = new HashMap<>();
+      if ("post".equalsIgnoreCase(method)) {
+        String postBody =
+            new String(
+                    ((ContentCachingRequestWrapper) request).getContentAsByteArray(),
+                    StandardCharsets.UTF_8)
+                .lines()
+                .collect(Collectors.joining(""));
+        JsonNode postBodyJson = objectMapper.readTree(postBody);
+        stackDriverPayload.put("postBody", postBodyJson);
+      }
+      // The warning here can be ignored, as StackDriver logging will interpret the extra arg
+      // and insert it into the `jsonPayload`. It is ignored when running locally.
```

Comment:
Can you describe in more detail how this works? Is it a side effect of using JSON logging? A reference to a doc would be nice, given that, from reading the doctoring to `Logger.info()`, it looks like an extra argument would be ignored.

Reply:
I took inspiration from here: https://github.com/DataBiosphere/terra-common-lib/blob/085750be68d67fed29b59302fa8b38fa944d6850/src/main/java/bio/terra/common/logging/RequestLoggingFilter.java#L92

Not sure how it works. Seems kinda magical to me.

Reply:
Hmm...also worth noting in there that they appear to put a limit of 10K characters on the request...might makes sense to do that too.

Reply:
OK, the key point to include in a comment here is that this depends on using a custom logback JSON layout, which is provided by the TCL. If you can think of a good way to do it, I would link directly to the class that implements this layout, since it has all the relevant information. https://github.com/DataBiosphere/terra-common-lib/blob/develop/src/main/java/bio/terra/common/logging/GoogleJsonLayout.java